src/Psalm/Config.php

```diff
@@ -2781,8 +2781,22 @@ public function getPHPVersionFromComposerJson(): ?string
                 $version_parser = new VersionParser();
 
                 $constraint = $version_parser->parseConstraints($php_version);
-
-                foreach (['5.4', '5.5', '5.6', '7.0', '7.1', '7.2', '7.3', '7.4', '8.0', '8.1'] as $candidate) {
+                $php_versions = [
+                    '5.4',
+                    '5.5',
+                    '5.6',
+                    '7.0',
+                    '7.1',
+                    '7.2',
+                    '7.3',
+                    '7.4',
+                    '8.0',
+                    '8.1',
+                    '8.2',
+                    '8.3',
+                ];
```

Comment:
Note to self: we keep forgetting to update PHP version numbers in different places. This means we need to centralize PHP version handling in a single place.